lib/SILOptimizer/IPO/GlobalOpt.cpp

```diff
@@ -770,6 +786,35 @@ void SILGlobalOpt::optimizeInitializer(SILFunction *AddrF,
   HasChanged = true;
 }
 
+/// Look through all the uses of the global variable. If the only use is a
+/// store, and the global is private, then we can remove the global, the store,
+/// and the alloc. Otherwise, return.
+void SILGlobalOpt::tryOptimizeUnusedGlobal(SILGlobalVariable *global,
+                                           StoreInst *store) {
+  if (global->getLinkage() != SILLinkage::Private)
```

Comment:
I think the canonical check is `isPossiblyUsedExternally(global->getLinkage(), module->isWholeModule())`, which should also handle hidden linkage in WMO builds